### PR TITLE
Better issue templates!

### PR DESCRIPTION
**Pull request by thecogdoctor**
General improvements to the default issue templates.
I've also removed the Question template as it's inclusion is not appropriate given questions can and should be asked on the ReSwitched discord. Issues are used for features, bugs and functionality, not misc or unneeded questions which are to be put through the proper channels.

- [X] Add system version prompt for bug reports

### CODE REVIEW

**Comment by misson20000** [comment]
Add "system firmware version" to bug report template please

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+### Crash Report
 
-(If a crash report was created under /atmosphere/crash_reports/, please upload it to 
-[gist](https://gist.github.com/) and paste the link here.)
+If your issue caused Atmosphère to crash, include the crash report(s) by creating a [gist](https://gist.github.com/) which you should the link here.
+Failure to include a crash report for crash related issues could mean your issue is closed with no notification or reason.
+
+### System Firmware Version
+X.X.X  
+[ Replace X's with system firmware version at time of crash. ]
```
Comment (roblabla) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ Replace X's with system firmware version at time of crash. ]
[ You can find your firmware version in the Settings -> System, under "System Update". ]
[ If it says "Update Pending", you can clear the pending update by rebooting to Maintenance Mode. ]
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -1,12 +1,46 @@
...
+### What component do you feel this would best fit within?
+- [Fusé](https://github.com/Atmosphere-NX/Atmosphere#components)
+  - Atmosphère's custom bootloader.
+- [Exosphère](https://github.com/Atmosphere-NX/Atmosphere#components)
+  - Fully-featured custom secure monitor.
+- [Stratosphère](https://github.com/Atmosphere-NX/Atmosphere#components)
+    - Custom system modules.
+- **[Thermosphère](https://github.com/Atmosphere-NX/Atmosphere#components)** ******
+  - Atmosphère's emuNAND implementation.
+- **[Troposphèr. ](https://github.com/Atmosphere-NX/Atmosphere#components)** ******
```
Comment (roblabla) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
- **[Troposphère. ](https://github.com/Atmosphere-NX/Atmosphere#components)** ******
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
 
-### Crash report?
+[ Include a detailed step by step process for recreating your issue. ]
+[ Note: If software you have installed can be used to make 'backups' of game files or distributed explicitly for piracy and are connected to why Atmosphère crashed, we will not provide support related to this issue. ]
+
+### Crash Report
 
-(If a crash report was created under /atmosphere/crash_reports/, please upload it to 
-[gist](https://gist.github.com/) and paste the link here.)
+If your issue caused Atmosphère to crash, include the crash report(s) by creating a [gist](https://gist.github.com/) which you should the link here.
```
Comment (roblabla) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ If your issue caused Atmosphère to crash, include the crash report(s) by creating a [gist](https://gist.github.com/) which you should the link here. ]
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -1,25 +1,44 @@
...
+Failure to include a crash report for crash related issues could mean your issue is closed with no notification or reason.
```
Comment (roblabla) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ Failure to include a crash report for crash related issues could mean your issue is closed with no notification or reason. ]
```

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (roblabla) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Should still mention the location of crash reports so users know where to find them.

**Comment by thecogdoctor** [comment]
@roblabla & @SciresM if you could give it a final review before merging for spelling and grammar that would complete this PR's review(s).

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -1,12 +1,46 @@
...
+  - [ If you are suggesting porting features or including features from other projects, include what license they are distributed under and what, if any libraries those project use. ]
+- Change
+- Removal
+  - [Ex: Removal of certain features or implementation due to a specific issue/bug or because of low quality code, etc.]
 
+[ Required: Reference the already existing [feature request. ](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]  
+[ Failure to do so means your request can and likely will be closed with no warning. ]
+
+### What component do you feel this would best fit within?
+- [Fusé](https://github.com/Atmosphere-NX/Atmosphere#components)
```
Comment (ThatNerdyPikachu) [inline .github/ISSUE_TEMPLATE/feature_request.md]: ``Fusé`` -> ``Fusée``

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -1,25 +1,47 @@
 ---
 name: Bug Report
 about: Something doesn't work correctly in Atmosphère.
-
+#assignees:
 ---
 
 ## Bug Report
 
+[ If any section does not apply, use nil. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: What does "use nil" mean? This can be made more clear.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -1,25 +1,47 @@
...
 ## Bug Report
 
+[ If any section does not apply, use nil. ]  
+[ Lines between [ ] (square brackets) should be removed before posting. ]
+
 ### What's the issue you encountered?
 
+[ Describe the issue in detail and what you were doing beforehand. ]  
+[ Did you make any changes related to Atmosphère itself? ]  
+[ If so make sure to include details relating to what exactly you changed. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ If so, make sure to include details relating to what exactly you changed. ]
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -1,25 +1,47 @@
...
+
 ### How can the issue be reproduced?
 
-### Crash report?
+[ Include a detailed step by step process for recreating your issue. ]
+[ Note: If software you have installed can be used to make 'backups' of game files or distributed explicitly for piracy and are connected to why Atmosphère crashed, we will not provide support related to this issue. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: This sentence is very hard to parse. It can definitely be cleaned up.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -1,25 +1,47 @@
...
+
+### Crash Report
 
-(If a crash report was created under /atmosphere/crash_reports/, please upload it to 
-[gist](https://gist.github.com/) and paste the link here.)
+[ Crash reports can be found under ``/atmosphere/crash_reports``. ]  
+[ If your issue caused Atmosphère to crash, include the crash report(s) by creating a [gist](https://gist.github.com/) which you should the link here.
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ If your issue caused Atmosphère to crash, include the crash report(s) by creating a [gist](https://gist.github.com/) and pasting the link here.
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 10]
```diff
@@ -1,25 +1,47 @@
...
+[ Failure to include a crash report for crash related issues could mean your issue is closed with no notification or reason. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I'd remove this line entirely. If someone posts an issue without a crash report, we'll just ask them for the crash report. No sense in closing the issue there. Maybe put a label on it to indicate that we're waiting on a crash report, but I don't think we have a high enough volume of issues for that to be a realistic concern.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 11]
```diff
@@ -1,25 +1,47 @@
...
+
+### System Firmware Version
+X.X.X  
+[ Replace X's with system firmware version at time of crash. ]
+[ You can find your firmware version in the Settings -> System, under "System Update". ]
+[ If it says "Update Pending", you can clear the pending update by rebooting to Maintenance Mode. ]
 
 ### Environment?
 
 - What bootloader (fusee, hekate, etc) was Atmosphère launched by:
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
- What bootloader (fusèe, hekate, etc) was Atmosphère launched by:
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 12]
```diff
@@ -1,25 +1,47 @@
...
-- Official release or unofficial build:
+- Official release or unofficial build: 
+  - [ Offical Release Version x.x.x (or) Unofficial Build ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
  - [ Offical release version x.x.x (or) unofficial build ]
```
Title casing doesn't make sense here, and capitalization should be consistent with the line before it.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 13]
```diff
@@ -1,25 +1,47 @@
...
+  - [ If using an unofficial build, include details on where/how you acquired the build. Self-compilation vs ex: Kosmos included version of Atmosphère ]
 - Do you have additional kips you're loading:
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Maybe change to "kips or sysmodules"

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 14]
```diff
@@ -1,25 +1,47 @@
...
-- Additional info about your environment:
+- Homebrew software installed:
+  - [ Note: If software you have installed can be used to make 'backups' of game files or distributed explicitly for piracy and are connected to why Atmosphère crashed, we will not provide support related to this issue. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: This sentence should be cleaned up, too.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 15]
```diff
@@ -1,25 +1,47 @@
...
 
 ### Additional context?
+- Additional info about your environment:
+- Any other information relevant to your issue.
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
- [ Any other information relevant to your issue. ]
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 16]
```diff
@@ -1,12 +1,45 @@
...
 name: Feature Request
-about: You want to suggest a new feature for Atmosphère.
-
+about: Suggest a new feature for Atmosphère.
+#assignees:
 ---
 
 ## Feature Request
-
+[ If any section does not apply, use nil. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]: Same thing here. "Use nil" is a strange expression.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 17]
```diff
@@ -1,12 +1,45 @@
...
+- [ These may include specific methods of implementation etc. ]
+
+#### Nature of Request:  
+[ Remove all that do not apply to your request. ]
+- Addition
+  - [ Ex: Addition of certain original features or features from other community projects. ]
+  - [ If you are suggesting porting features or including features from other projects, include what license they are distributed under and what, if any libraries those project use. ]
+- Change
+- Removal
+  - [Ex: Removal of certain features or implementation due to a specific issue/bug or because of low quality code, etc.]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]: I really don't think we need three separate sections for describing a feature request. Just a "feature description" section would be enough.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 18]
```diff
@@ -1,12 +1,45 @@
...
 
+[ Required: Reference the already existing [feature request. ](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]: We can phrase this more politely.
```suggestion
[ Note: Please [search for previous feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) to see if your idea has already been suggested before opening a new issue.]  
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 19]
```diff
@@ -1,12 +1,45 @@
...
+[ Remove all that do not apply to your request. ]
+- Addition
+  - [ Ex: Addition of certain original features or features from other community projects. ]
+  - [ If you are suggesting porting features or including features from other projects, include what license they are distributed under and what, if any libraries those project use. ]
+- Change
+- Removal
+  - [Ex: Removal of certain features or implementation due to a specific issue/bug or because of low quality code, etc.]
 
+[ Required: Reference the already existing [feature request. ](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]  
+[ Failure to do so means your request can and likely will be closed with no warning. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]: Again, I don't think this is actually a reasonable thing for us to do, even if the user didn't bother to search for open feature requests.
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 11 above)
Comment (misson20000) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Err, please double check that I didn't botch the formatting by accidentally removing that space.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Nil simply means that it's a not an applicable section. Will change to N/A.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 13 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Copied from the original template, will include.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 19 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: It should fall upon the user to consult the open feature requests. It is your job as to developer to write code rather than deal with duplicate issues.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 18 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Politeness while ideal, is not the reality. It should be a requirement to reference existing feature requests before a user makes a potentially duplicate issue.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 10 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I will be revising and removing the line saying that failure to include a crash report will cause your issue to be closed, but I will maintain the fact that the user needs to include a crash report.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 17 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: The reasoning behind the 3 sections is too weed out generic and low effort requests. I do agree it could be condensed but this section was made based upon my experiences with other repositories and contributions.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 18 above)
Comment (misson20000) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Sure, but I think that my suggested changes convey the same idea more clearly and without coming off as harshly. We (or at least I) want to be somewhat inviting. I think the kind of people who see "Note: Please search for previous feature requests" and think it's less mandatory than "Required: Reference the already existing feature request" aren't the kind of people who would do a good job of searching for that anyway.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 19 above)
Comment (misson20000) [reply .github/ISSUE_TEMPLATE/feature_request.md]: It really should, but if users don't check for open feature requests, I believe that it's our tasks as *maintainers* to keep the open issue list clean and provide justification for why we're closing issues. This phrasing is uninviting and implies that we're going to be closing issues without any clarification or justification.

File: .github/ISSUE_TEMPLATE/question.md [hunk 20]
```diff
@@ -1,12 +1,7 @@
...
-
-If you have a question, please join the ReSwitched discord for help.
-
-- Discord link: https://discordapp.com/invite/DThbZ7z
-
+about: Please ask questions in the ReSwitched discord, instead of making issues. 
+--- 
+We would like to use GitHub to keep track of problems/feature requests.  
+If you have a question, please join the ReSwitched discord for help. 
+- Discord link: https://discordapp.com/invite/DThbZ7z
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/question.md]: Perhaps we should put a badge like this on the README: [Chat on Discord](https://discordapp.com/invite/ZdqEhed)

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 19 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: I will definitely rephrase it to something that would inspire confidence in potential contributors. Cheers.

File: .github/ISSUE_TEMPLATE/question.md (hunk 20 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/question.md]: That would be a much better solution rather than have the issue.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 8 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed with new template for support disclaimers at the top of bug reports by default.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 14 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed with new support disclaimer at the top of the issue template.

File: .github/ISSUE_TEMPLATE/question.md (hunk 20 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/question.md]: Added a the provided discord badge to README.md

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 19 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Addressed and rephrased.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 18 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Addressed and rephrased.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 21]
```diff
@@ -1,25 +1,57 @@
...
 
+[ If any section does not apply, use N/A. ]  
+[ Lines between [ ] (square brackets) should be removed before posting. ]
+
+[ * ]  
+[ Note: If the bug or crash you encountered is related to; ]  
+[ - software used to make "backups", ]  
+[ - software explicitly distributed for piracy, ]  
+[ - other software related to the duplication of copyrighted works,  
+no support for your issue will be provided and your issue will be closed. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Sorry about the nitpick, but the way this line is wrapped is inconsistent with the rest of the document. Everywhere else, lines aren't wrapped. If you think this line should be wrapped, should it have an individual set of brackets for each line?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 22]
```diff
@@ -1,25 +1,57 @@
 ---
 name: Bug Report
 about: Something doesn't work correctly in Atmosphère.
-
+#assignees:
 ---
 
 ## Bug Report
 
+[ If any section does not apply, use N/A. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
[ If any section does not apply, replace its contents with "N/A". ]
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 23]
```diff
@@ -1,12 +1,47 @@
...
 name: Feature Request
-about: You want to suggest a new feature for Atmosphère.
-
+about: Suggest a new feature for Atmosphère.
+#assignees:
 ---
 
 ## Feature Request
 
+[ If any section does not apply or you do not have the information needed for that section, use N/A (Not Applicable). ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
[ If any section does not apply, replace its contents with "N/A". ]
[ If you do not have the information needed for a section, replace its contents with "Unknown". ]
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 24]
```diff
@@ -1,12 +1,47 @@
...
+about: Suggest a new feature for Atmosphère.
+#assignees:
 ---
 
 ## Feature Request
 
+[ If any section does not apply or you do not have the information needed for that section, use N/A (Not Applicable). ]  
+[ Lines between [ ] (square brackets) are to be removed before posting. ]
+
+[ Please refer to the already existing [feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]
```
Comment (misson20000) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
[ Please search for existing [feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22features%2Ffeature-request%22) before you make your own request. ]
```
Remove the `is: open` filter. We don't want duplicates of closed feature requests.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 21 above)
Comment (thecogdoctor) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I fiddled around with it and decided I had no idea what I wanted. I'll fully wrap both lines just to be careful.

**Comment by thecogdoctor** [comment]
@misson20000 extraneous white-space, particularly 2 spaces at the end of a line, is what GFM uses to distinguish true line-endings. It's dumb and I agree, but it needs to remain to ensure formatting isn't broken.

**Comment by SciresM** [comment]
@thecogdoctor extraneous whitespace isn't needed at the end of lines. If formatting is broken, add additional newlines.

**Comment by thecogdoctor** [comment]
> 
> 
> @thecogdoctor extraneous whitespace isn't needed at the end of lines. If formatting is broken, add additional newlines.

I might be mistaken but when I remove the 2 trailing white-spaces from these lines from question.md
```
---
name: Question
about: Please ask questions in the ReSwitched discord, instead of making issues.
---
We would like to use GitHub to keep track of problems/feature requests.
If you have a question, please join the ReSwitched discord for help.
- Discord link: https://discordapp.com/invite/DThbZ7z
```
With whitespace (the small dots):
Whitespace
Without trailing whitespace:
No-Whitespace

(Note: These previews were generated with atom's markdown preview plugin.)

**Comment by thecogdoctor** [comment]
I've added proper line-breaks with the previous commits, removing the extraneous whitespace as specified by @misson20000, cheers. 👍
